--- a/config/template.cmake
+++ b/config/template.cmake
@@ -4,8 +4,6 @@
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}")
 
 if(NOT TARGET "@PROJECT_NAME@::@PROJECT_NAME@")
-  include("${CMAKE_CURRENT_LIST_DIR}/@PROJECT_NAME@-targets.cmake")
-
   include(CMakeFindDependencyMacro)
 
   if(NOT TARGET "OpenMP::OpenMP_Fortran" AND "@PROJECT_NAME@_WITH_OpenMP")
@@ -13,10 +11,12 @@
   endif()
 
   if(NOT TARGET "LAPACK::LAPACK")
-    find_dependency("LAPACK")
+    find_dependency("Lapack")
   endif()
 
   if(NOT TARGET "mctc-lib::mctc-lib")
     find_dependency("mctc-lib")
   endif()
+
+  include("${CMAKE_CURRENT_LIST_DIR}/@PROJECT_NAME@-targets.cmake")
 endif()
