--- a/cMake/FreeCAD_Helpers/SetupQt.cmake
+++ b/cMake/FreeCAD_Helpers/SetupQt.cmake
@@ -14,6 +14,10 @@
         list (APPEND FREECAD_QT_COMPONENTS GuiTools)
         list (APPEND FREECAD_QT_COMPONENTS SvgWidgets)
         list (APPEND FREECAD_QT_COMPONENTS OpenGLWidgets)
+        if(WIN32)
+          set(QT_NO_PRIVATE_MODULE_WARNING ON)
+          list(APPEND FREECAD_QT_COMPONENTS GuiPrivate)
+        endif()
     endif()
 
     list (APPEND FREECAD_QT_COMPONENTS OpenGL PrintSupport Svg UiTools Widgets LinguistTools)
